cmd/snap-confine/spread-tests/main/cgroup-used/task.yaml

```diff
@@ -8,19 +8,15 @@ execute: |
     cd /
     echo "Clear udev tags and cgroups with non-test device and running snapd-hacker-toolbelt.busybox"
     echo 'KERNEL=="uinput", TAG+="snap_snapd-hacker-toolbelt_busybox"' > /etc/udev/rules.d/70-spread-test.rules
-    udevadm control --reload-rules
-    udevadm settle
-    udevadm trigger
+    udevadm trigger --action=change
     udevadm settle
```

Comment:
i would have done it exactly the other way around, keep the "control --reload-rules" and drop the artificial trigger call (which will trigger a complete set of coldplug events for everything). the --reload-rules should actually execute coldplug for new found rules automatically ... (also, according to the manpage --action=change is now the default, but i dont know if thats a debian/ubuntu thing or an upstream change applying to all distros we could run on)

Reply:
@ogra1 - why? udev should pick up the rules automatically and we want to call trigger because we want the rules to take effect when connecting the interfaces, so the trigger is necessary for existing devices.

Reply:
Also, the actions that can be specified are add, remove and change. AUIU, running 'change' should be safe because it won't add then remove devices, so ordering shouldn't change. Instead, it'll look at the rules in its db and update devices as needed-- in our case, by adding tags.

Can you more clearly show that 'change' is dangerous? All I've found supports it is safe. Indeed, we've used trigger since 15.04 and never seen a bug related to the HDD coming up differently (as you mentioned in the forum).